src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -283,6 +283,27 @@ private ReadOnlyMemory<byte> GetPropertyRawValue(int valueIndex)
                 : JsonReaderHelper.TranscodeHelper(segment);
         }
 
+        internal ReadOnlySpan<byte> GetRawUtf8Span(int index, JsonTokenType expectedType)
+        {
+            CheckNotDisposed();
+
+            DbRow row = _parsedData.Get(index);
+
+            JsonTokenType tokenType = row.TokenType;
+
+            if (tokenType == JsonTokenType.Null)
+            {
+                return default;
+            }
+
+            CheckExpectedType(expectedType, tokenType);
```

Comment:
Given that we only call this method internally, couldn't this have been a debug assert instead, i.e. put the following at the top of the method?
```C#
Debug.Assert(JsonTokenType.PropertyName, row.TokenType);
```

Reply:
Sure. I didn't notice `JsonDocument.GetString` uses `CheckExpectedType` because it has a public access path through `JsonElement.GetString`.